lib/android-helpers.js

```diff
@@ -797,7 +797,11 @@ helpers.initDevice = async function initDevice (adb, opts) {
   if (skipDeviceInitialization) {
     logger.info(`'skipDeviceInitialization' is set. Skipping device initialization.`);
   } else {
-    await adb.waitForDevice();
+    if (helpers.isEmulator(adb, opts)) {
```

Comment:
I assume this was the case for emulators like bluestacks, where we have no reliable method of detecting whether this is a real device.

Reply:
this condition could be combined into `else` clause above (e.g. `else if`)

Reply:
pushSettingsApp in the next method call is necessary for `if (helpers.isEmulator(adb, opts))` case as well